File: main_project/client/java/OceAdapter/src/main/java/com/xiaonei/xce/pagefans/PageFansReloaderAdapter.java
package com.xiaonei.xce.pagefans;

import xce.buddy.PageFansReloaderPrx;
import xce.buddy.PageFansReloaderPrxHelper;
import xce.clusterstate.ReplicatedClusterAdapter;

public class PageFansReloaderAdapter extends ReplicatedClusterAdapter{


	private static final String ctr_endpoints = "ControllerPageFansReloader";	
	private static final int _interval = 120;

	public PageFansReloaderAdapter() {
		super(ctr_endpoints, _interval);
		
	}
			
	protected PageFansReloaderPrx getReloaderPrx(int userId) {
		Ice.ObjectPrx prx0 = getValidProxy(userId,300);
		if (prx0 != null) {
			PageFansReloaderPrx prx = PageFansReloaderPrxHelper
					.uncheckedCast(prx0.ice_timeout(300).ice_twoway());
			return prx;
		} else {
			return null;
		}
	}
  
	protected PageFansReloaderPrx getReloaderOnePrx(int userId) {
		Ice.ObjectPrx prx0 = getValidProxy(userId,300);
		if (prx0 != null) {
			PageFansReloaderPrx prx = PageFansReloaderPrxHelper
					.uncheckedCast(prx0.ice_timeout(300).ice_oneway());
			return prx;
		} else {
			return null;
		}
	}
	

	

}

File: main_project/OceCxxAdapter/src/PageFansReloaderAdapter.cpp
#include "PageFansReloaderAdapter.h"
#include "PageFans.h"
 
using namespace xce::buddy;

PageFansReloaderAdapter::PageFansReloaderAdapter() : MyUtil::ReplicatedClusterAdapterI <PageFansReloaderPrx>("ControllerPageFansReloader", 120, 300){
}

void PageFansReloaderAdapter::reload(int id)
{
		try {
				getProxy(0)->reload(id);
		} catch (const Ice::Exception& e) {
				MCE_WARN("[PageFansReloaderAdapter::addFan] Ice::Exception " << e);
		}
		return ;


}


